src/util/ringdelaybuffer.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include "util/samplebuffer.h"
+#include "util/types.h"
+
+// TODO(davidchocholaty) Done documentation.
+
+/// The ring delay buffer allows moving with the reading position subject
+/// to certain rules. Another difference between the classic ring buffer is,
+/// that the ring delay buffer offers to read zero values
+/// which were not written by using the write method and write position.
+/// Both of these two specific properties are based on the cross-fading
+/// between changes of two delays.
+class RingDelayBuffer final {
+  public:
+    RingDelayBuffer(SINT bufferSize);
```

Comment:
Ideally this would be generic, but that would essentially make this about as complex as `std::vector` with all the required SFINAE (or C++20 concepts work)...